sway-lsp/src/core/session.rs

```diff
@@ -435,13 +462,60 @@ pub fn parse_project(
     Ok(())
 }
 
+/// Parse the [LexedProgram] to populate the [TokenMap] with lexed nodes.
+pub fn parse_lexed_program(
+    lexed_program: &LexedProgram,
+    ctx: &ParseContext,
+    modified_file: &Option<PathBuf>,
+    f: impl Fn(&Annotated<ItemKind>, &ParseContext) + Sync,
+) {
+    let should_process = |item: &&Annotated<ItemKind>| {
+        modified_file
```

Comment:
This could probably be abstracted out to a helper function given it's repeated in the various `parse_X()` functions

Reply:
For some quick wins that's a huge improvement 🚀

Reply:
Thanks Alfie, I might do this in a follow up.